cpp/src/arrow/compute/api_vector.h

```diff
@@ -188,6 +188,29 @@ class ARROW_EXPORT PartitionNthOptions : public FunctionOptions {
   NullPlacement null_placement;
 };
 
+/// \brief Options for IsMonotonic
+class ARROW_EXPORT IsMonotonicOptions : public FunctionOptions {
+ public:
+  enum NullHandling {
+    // Ignore nulls.
```

Comment:
Please use /// to allow doxygen to detect docstrings
```suggestion
    /// Ignore nulls.
```